bin/_main

```diff
@@ -508,20 +512,27 @@ elif [ "${1}" == "run" ]; then
     fi
 
     tool_params_file=${base_run_dir}/config/tool-params.json
-    if [ ! -e ${tool_params} ]; then
-        cp $rs_dir/config/tool-params.json ${tool_params_file}
-        echo "You do not have a \"tool-params.json\" in the current directory and have not explicitly"
-        echo "specified where to find a tool parameters file with \"--tools-params=<file>\"."
-        echo ""
-        echo "Crucible will use the default tools found in $rs_dir/config/tool-params.json:"
-        cat ${tool_params_file}
-        echo ""
-        echo "If you wish to use different tools, create a tool parameters file which adheres to"
-        echo "the proper schema ($rs_dir/schema/tools/json) and either place that file in the"
-        echo "current directory named \"tool-params.json\" or explicitly specify it with"
-        echo "\"--tool-params=<file>\"."
+    if [ "${no_tools}" == "0" ]; then
+        if [ ! -e ${tool_params} ]; then
+            cp $rs_dir/config/tool-params.json ${tool_params_file}
+            echo "You do not have a \"tool-params.json\" in the current directory and have not explicitly"
+            echo "specified where to find a tool parameters file with \"--tools-params=<file>\"."
+            echo ""
+            echo "Crucible will use the default tools found in $rs_dir/config/tool-params.json:"
+            cat ${tool_params_file}
+            echo ""
+            echo "If you wish to use different tools, create a tool parameters file which adheres to"
+            echo "the proper schema ($rs_dir/schema/tools/json) and either place that file in the"
+            echo "current directory named \"tool-params.json\" or explicitly specify it with"
+            echo "\"--tool-params=<file>\"."
+            echo ""
+            echo "You can also disable tools entirely using the \"--no-tools\" parameter."
+        else
+            cp ${tool_params} ${tool_params_file}
+        fi
     else
-        cp ${tool_params} ${tool_params_file}
+        echo "Disabling all tools due to --no-tools"
+        echo "[]" > ${tool_params_file}
```

Comment:
So .. we will have this behavior ... there is a "custom" tools-params.json in-place, but the user just want to run --no-tools for this one run for whatever reason. Once the --no-tools run runs, the original tools-params.json will be overwritten with "[]".  Is that right?

Reply:
Yes.  The tool-params.json that is created via the run command (either via copying the default set or by this new method) is always per run.  There is no persistence from run-to-run -- besides whatever files the user themselves has created in their directory/directories.

Reply:
So your original tool params file is safe/unmodified